### PR TITLE
Create test for installing and enrolling an agent

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do?

Closes https://github.com/elastic/elastic-agent/issues/2402

This adds a new test that installs and enrolls an agent, checks for logs, checks for errors, then unenrolls the agent. This also adds a series of elasticsearch helpers that can be used by other tests.

Note that my Elasticsearch skills are a little rusty, so would appreciate someone with more ES skills looking over the queries.

This also fixes a few issues with the FQDN test.

## Why is it important?

We want more test coverage.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

## Running the test

- NOTE: setting `TEST_INTEG_AUTH_GCP_DATACENTER` can speed things up. See https://gcping.com/ to see what the closest datacenter is to you.
- run `mage integration:single TestEnrollAndLog`

### CODE REVIEW

File: testing/integration/enroll_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	docs, err = tools.FindMatchingLogLines(runner.requirementsInfo.ESClient, "metricbeat start running")
+	require.NoError(runner.T(), err)
+	require.NotZero(runner.T(), len(docs.Hits.Hits))
+
+	docs, err = tools.FindMatchingLogLines(runner.requirementsInfo.ESClient, "filebeat start running")
+	require.NoError(runner.T(), err)
+	require.NotZero(runner.T(), len(docs.Hits.Hits))
+
+	// Stage 6: Unenroll
+	err = tools.UnEnrollAgent(runner.requirementsInfo.KibanaClient)
```
Comment (blakerouse) [inline testing/integration/enroll_test.go:126]: Should add this in a `t.Cleanup` that way its always done even if something in the test fails. Just to keep the stack clean.

File: testing/integration/fqdn_test.go [hunk 2]
```diff
@@ -103,8 +103,14 @@ func (s *FQDN) TestFQDN() {
 			kibana.MonitoringEnabledLogs,
 			kibana.MonitoringEnabledMetrics,
 		},
+		AgentFeatures: []map[string]interface{}{
+			{
+				"name":    "fqdn",
+				"enabled": false,
+			},
+		},
```
Comment (blakerouse) [inline testing/integration/fqdn_test.go:111]: Why this change? What does it fix? Is `enabled: false` not the default?

**Review by blakerouse — APPROVED** [review]
Overall I think this looks really good. I like all the added helpers that will be very useful for so many tests. I am no expert at elasticsearch queries, what I read looked correct to me.

I only added to comments, but no blockers. +1 from me!

File: testing/integration/fqdn_test.go (hunk 2 above)
Comment (fearful-symmetry) [reply testing/integration/fqdn_test.go:111]: So, at least on earlier iterations of the Kibana helper libraries, the API call would fail if `AgentFeatures` was empty.

File: testing/integration/fqdn_test.go [hunk 3]
```diff
@@ -103,8 +103,14 @@ func (s *FQDN) TestFQDN() {
...
 	}
-	policy, err := tools.InstallAgentWithPolicy(s.T(), s.agentFixture, kibClient, createPolicyReq)
+	policy, err := tools.InstallAgentWithPolicy(s.T(), s.agentFixture, kibClient, s.requirementsInfo.ESClient, createPolicyReq)
```
Comment (ycombinator) [inline testing/integration/fqdn_test.go:113]: It looks like this change broke the build on `main`:

https://buildkite.com/elastic/elastic-agent/builds/1394#01890d53-55b0-4225-8786-35bb175ef164/56-241

Once https://github.com/elastic/elastic-agent/pull/2928 is merged, CI will be able to catch such issues on PRs before they're merged into `main`.
